### ISSUE TITLE
web based association analysis

### ISSUE THREAD

**Issue by BoPeng** [vatlab/VAT#4]
We proposed to provide web based association analysis method. Given the nature of association analysis, I assume that the server should have all the reference data (1000G etc) and annotations that can be readily usable.

Users can upload their data to the server, through an interface similar to basespace (illumina). They can then select data to use (their own data as well as public reference data), phenotype to analyze, and optionally statistically method. The VAT server would then perform the analysis and provide results through the web interface.

Given the difficulty to set up infrastructure (e.g. hadoop/spark) for large scale analysis, I assume that asking users to upload their data would be easier than asking them to set up the infrastructure and prepare all the reference data.

**Comment by gaow** [comment]
Users may not want to upload the data for security concerns. Also there are a lot to be done before association analysis, many of which are interactive and iterative depending on judgement calls by the analyst based on what they see at each step. On top of that there is the computational capacity we can afford to offer. Therefore the web-based analysis is a bit ambitious to me.

Also although we call the project VAT, a very useful application is still variant filtering, for which `gemini` seems to be taking the lead currently, although it does not necessarily do more than what we can offer now. It would be a pity if we limit ourselves to association analysis only, considering what we already have in version 2X. Variant filtering is very much interactive. I can only imaging a proper web interface being something like a Jupyter notebook that keeps track of all the attempts and diagnostic figures -- there is no one-click button to get it work. 

That said, I do agree setting up the infrastructure and annotations can be a pain. But for the scale in version 2X the `--update-resource` is still acceptable. Perhaps we can work on downloading annotations for only the variants involved, thus avoid downloading WGS annotations for WES projects? Then they'll have to re-download whenever new variants are added, though.

**Comment by BoPeng** [comment]
The whole idea of web-based analysis is to free users from making choices between 0.05 and 0.01 cutoff, namely we are supposed to make the filtering decisions for them (albeit allowing some flexibility). A majority of users have no idea the impact of choice of parameters and it should be up to us to decide what method and parameters to use.

Regarding HIPPA compliance and computational capacity, I am not saying that we are providing this service right now. If Galaxy and basespace can surpass HIPPA, we can see what we can do for it. It is doable but we need to provide the required security and policy. Also, existing funding would allow us to rent resources from amazon cloud and set up hadoop cluster from which we can develop our service. Once the service is more or less complete (even with just a rough proof-of-idea implementation), we can submit another R01 for support, and/or provide fee-based service to cover cloud cost (which will be hard).

I am not proposing that we give up standalone vtools/vat and vat will continue to provide filtering features.  Advanced users can certainly download vat, use them for smaller projects, create their own infrastructure for larger ones, or modifying the underlying (SoS) pipeline to accommodate their own methods. The idea is that we should have more ambitious plans in mind and consider these plans when we design the architecture of vat.

**Comment by gaow** [comment]
I agree we should be ambitious in planning. I was just worried VAT will have to limit what it does in order to work as a web app.

For efforts at this scale I assume we'll have short-term, mid-term and long-term action plans. Perhaps after some more brainstorming we can create wiki pages to document these plans.

**Comment by BoPeng** [comment]
Just for the record of a more ambitious plan for a future grant application, it is possible for us to become an association analysis warehouse to which data from different sources are deposited, and more associations could be mined from an increasing amount of data. The data could be from genetic epidemiological data that have been analyzed, or even required by NIH to deposit.

This would require that we have a truly distributed storage and data service scheme. a reference-genome neutral variant system, a data dictionary (ontology) to ensure consistency of genes and phenotypes from different sources.

**Comment by gaow** [comment]
If we have made good progress with SoS on cluster, perhaps we can just create HTML interface for VAT and find a way to install the interface to a cluster at the user's end, and submit / monitor SoS jobs for VAT. I recall it is the plan to have some progress monitoring for SoS. I think the efforts here on SoS and VAT can merge.

**Comment by BoPeng** [comment]
SoS with the current design would not work well for the bigger plan we have in mind for VAT. Let us keep them separate for now.
